app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tab.kt

```diff
@@ -0,0 +1,396 @@
+  tabTitles: List<String>,
+  selectedTabIndex: Int,
+  onTabChosen: (index: Int) -> Unit,
+  modifier: Modifier = Modifier,
+  tabSize: TabSize = TabDefaults.defaultSize,
+  tabStyle: TabStyle = TabDefaults.defaultStyle,
+) {
+  val density = LocalDensity.current
+  val currentWidthMap = remember { mutableStateMapOf<Int, Int>() }
+  val currentHeightMap = remember { mutableStateMapOf<Int, Int>() }
+  val currentOffsetMap = remember { mutableStateMapOf<Int, Offset>() }
+  val indicatorOffsetX: Dp by animateDpAsState(
+    targetValue = calculateIndicatorOffsetX(currentOffsetMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  val indicatorOffsetY: Dp by animateDpAsState(
+    targetValue = calculateIndicatorOffsetY(currentOffsetMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  val indicatorWidth: Dp by animateDpAsState(
+    targetValue = calculateIndicatorWidth(currentWidthMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  val indicatorHeight: Dp by animateDpAsState(
+    targetValue = calculateIndicatorHeight(currentHeightMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  Box(
+    modifier = modifier
+      .clip(tabSize.rowShape)
+      .background(tabStyle.colors.tabRowBackground)
+      .height(intrinsicSize = IntrinsicSize.Min)
+      .padding(tabSize.rowPadding),
+  ) {
+    TabIndicator(
+      indicatorHeight = indicatorHeight,
+      indicatorWidth = indicatorWidth,
+      indicatorOffsetX = indicatorOffsetX,
+      indicatorOffsetY = indicatorOffsetY,
+      indicatorColor = tabStyle.colors.chosenTabBackground,
+      indicatorShape = tabSize.tabShape,
+    )
+    FlowRow(
+      horizontalArrangement = Arrangement.Start,
+      verticalArrangement = Arrangement.Center,
+      maxItemsInEachRow = maxItemsInTheRow,
+      maxLines = maxLines,
+      overflow = FlowRowOverflow.Visible,
+      modifier = Modifier
+        .clip(tabSize.rowShape)
+        .fillMaxWidth(),
+    ) {
+      tabTitles.forEachIndexed { index, title ->
+        TabItem(
+          modifier = Modifier
+            .clip(tabSize.tabShape)
+            .weight(1f)
+            .onPlaced {
+              currentOffsetMap[index] = it.positionInParent()
+            },
+          onClick = {
+            onTabChosen(index)
+          },
+          text = title,
+          textStyle = tabSize.textStyle,
+          tabTextColor = tabStyle.colors.textColor,
+          contentPadding = tabSize.tabPadding,
+          onTextMeasured = {
+            currentWidthMap[index] = it
+          },
+          onHeightMeasured = {
+            currentHeightMap[index] = it
+          },
+        )
+      }
+    }
+  }
+}
+
+private fun calculateIndicatorWidth(map: SnapshotStateMap<Int, Int>, selectedTabIndex: Int, density: Density): Dp {
+  return with(density) {
+    map[selectedTabIndex]?.toDp() ?: 5.dp
```

Comment:
The 5f fallback, do we ever expect to hit that? is it for the first frame where the map may not have been populated yet?
Does this actually have any visual effect in any case or is it just so that it does not crash? Also if we do just `!!` does it crash? I am mostly curious, I don't really suggest we do `!!`

Reply:
`!!` just crashes. 5f (or 0f, doesn't matter) does give initial animation (the indicator expands to the element width and height from the top left corner), but I don't see an easy way to avoid that. Wdyt?

Reply:
Hmmmm alright so it is more tricky than I had hoped for initially.
So if we did something like we've done in places like these https://github.com/HedvigInsurance/android/blob/7739da9bd86dc664f8e479b4e11e0de17f79327c/app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Button.kt#L72-L80 do you imagine this working?
So in a box, layout the indicator as-is, but also layout the first entry in the flowrow too, but without placing it at all, only to have it take the same size.
so that the default size starts as the same as the first item anyway.

And you can then have this return null if it does not find anything in the map, and if that is the case just do not pass the width/height modifier at all, smth like
```
.then(
  if (resultFromInsideTheMap != null) {
    Modifier.width(resultFromInsideTheMap)
  } else {
    Modifier
  },
)
```

Does this make any sense at all?

Reply:
Great, will try that!